addons/udes_stock/models/stock_picking.py

```diff
@@ -681,3 +682,57 @@ def _unassign_users(self, skip_users=None):
         users_to_unassign.sudo().write(
             {"u_picking_id": False, "u_picking_assigned_time": False}
         )
+
+    @staticmethod
+    def get_stock_investigation_message(quants):
+        """
+        Dummy method to be overridden on a per-customer basis.
+
+        :returns: str
+        """
+        return ""
+
+    def raise_stock_investigation(self, reason, quants, location):
+        """
+        Raises a stock investigation picking with the provided reason, quants and location.
+        Un-reserves stock from affected pickings, creates stock investigation and reserves the stock,
+        finally tries to reserve stock for the original pickings.
+
+        :args:
+            - reason: str
+                The reason for which the stock is being investigated
+            - quants: stock.quant
+                The stock quants to reserve and investigate in the new picking
+            - location:
+                The location where the quants need to be investigated
+        """
+        Group = self.env["procurement.group"]
+
+        wh = self.env.user.get_user_warehouse()
+        stock_inv_pick_type = wh.u_stock_investigation_picking_type or wh.int_type_id
+        group = Group.get_or_create(reason, create=True)
+
+        # Post message and un-reserve stock
+        for pick in self:
+            pick.message_post(body=reason)
+            pick.do_unreserve()
+
+        # create a stock investigation pick
+        investigation_picking = quants.create_picking(
+            stock_inv_pick_type,
+            only_available=True,
+            location_id=location.id,
+            group_id=group.id,
+            confirm=True,
+            assign=True,
+        )
+
+        picking_details = self.get_stock_investigation_message(quants.exists())
+
+        if picking_details:
+            picking_header = "Stock Investigation created with: <br>"
+            investigation_picking.message_post(body=_(picking_header + picking_details))
```

Comment:
I would probably leave the translation of `pick_details` to `get_stock_investigation_message` and do something like:
```
picking_header = "Stock Investigation created with: <br>"
msg_body = _(picking_header) + pick_details
investigation_picking.message_post(body=msg_body)
```

Reply:
I'm not sure I understand? What are you suggesting here?